torch/_ops.py

```diff
@@ -336,6 +349,11 @@ def handler(*args, **kwargs):
             return handler
 
         key = resolve_key(self, key)
+
+        if key is torch._C.DispatchKey.Autograd:
```

Comment:
You need to interpose all of the other Autograd keys. This "accidentally" works here because of an infelicity in the python dispatcher which is that you never see fallthrough keys. If I register a per-backend Autograd key, you will see it show up in the dispatcher.

```
import torch.library

mylib = torch.library.Library("mylib", "DEF")

mylib.define("myop(Tensor self) -> Tensor")
def f(x):
    return x + 1
mylib.impl("myop", f, "AutogradCPU")

x = torch.randn(3)
with torch._dispatch.python.enable_python_dispatcher():
    torch.ops.mylib.myop(x)
```

when I add

```
diff --git a/torch/_ops.py b/torch/_ops.py
index b3ebd401ab..55a88ea0aa 100644
--- a/torch/_ops.py
+++ b/torch/_ops.py
@@ -337,7 +337,8 @@ class OpOverload(PyOperatorABC):
 
         key = resolve_key(self, key)
         r = self.py_kernels.get(key, key)
-        setattr(self, attr, r)
+        print(self, key)
+        # setattr(self, attr, r)
         return r
 
     def name(self):
```

I get printout

```
mylib.myop.default DispatchKey.AutogradCPU
aten.add.Tensor DispatchKey.Autograd
aten.add.Tensor DispatchKey.CPU
aten.to.dtype DispatchKey.CompositeImplicitAutograd
aten._to_copy.default DispatchKey.BackendSelect
aten._to_copy.default DispatchKey.CompositeExplicitAutograd
aten.empty_strided.default DispatchKey.BackendSelect
aten.copy_.default DispatchKey.CompositeExplicitAutograd
```

Reply:
Are you trying to support the case of overriding AutogradCPU impl dynamically for ops registered through torch.library? And for aten ops with backend specific Autograd functions, e.g AutogradXLA? 

```
import torch.library
from torch._dispatch.python import enable_python_dispatcher

mylib = torch.library.Library("mylib", "DEF")

mylib.define("myop(Tensor self) -> Tensor")
def f(x):
    return x + 1
my_op = mylib.impl("myop", f, "AutogradCPU")


def f2(x):
    return x * 2

# override AutogradCPU impl dynamically, and this should work? 
python_dispatch_table = {
    (torch.ops.mylib.myop.default, torch._C.DispatchKey.AutogradCPU): f2
}

x = torch.randn(3)
with enable_python_dispatcher(), torch._ops.python_dispatch(python_dispatch_table):
    torch.ops.mylib.myop(x)
```

Reply:
No, I just needed to make an op with the custom api to trigger this situation, as I don't think we have any ops in stock pytorch that have an AutogradCPU key. XLA, however, does have these